<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>glaxnimate::gui::ScriptConsole</class>
 <widget class="QWidget" name="ScriptConsole">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>249</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTextEdit" name="console_output">
     <property name="font">
      <font>
       <family>Monospace</family>
      </font>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="HistoryLineEdit" name="console_input">
       <property name="font">
        <font>
         <family>Monospace</family>
        </font>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="btn_clear_console">
       <property name="toolTip">
        <string>Clear Output</string>
       </property>
       <property name="text">
        <string>Clear</string>
       </property>
       <property name="icon">
        <iconset theme="edit-clear-all">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="check_persist">
       <property name="toolTip">
        <string>Persist Logs</string>
       </property>
       <property name="text">
        <string>Persist</string>
       </property>
       <property name="icon">
        <iconset theme="pin"/>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="btn_reload">
       <property name="toolTip">
        <string>Reload Script Modules</string>
       </property>
       <property name="text">
        <string>Reload</string>
       </property>
       <property name="icon">
        <iconset theme="view-refresh"/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="console_language"/>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HistoryLineEdit</class>
   <extends>QLineEdit</extends>
   <header>history_line_edit.hpp</header>
   <slots>
    <signal>lineExecuted(QString)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>console_input</sender>
   <signal>lineExecuted(QString)</signal>
   <receiver>ScriptConsole</receiver>
   <slot>console_commit(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>129</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_clear_console</sender>
   <signal>clicked()</signal>
   <receiver>ScriptConsole</receiver>
   <slot>console_clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>187</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>console_commit(QString)</slot>
  <slot>console_clear()</slot>
 </slots>
</ui>
